integration_tests/test_mempool.py

```diff
@@ -0,0 +1,87 @@
+from pathlib import Path
+
+import pytest
+from web3 import Web3
+
+from .network import setup_custom_cronos
+from .utils import (
+    ADDRS,
+    CONTRACTS,
+    KEYS,
+    deploy_contract,
+    sign_transaction,
+    wait_for_new_blocks,
+)
+
+pytestmark = pytest.mark.mempool
+
+
+@pytest.fixture(scope="module")
+def cronos(tmp_path_factory):
+    path = tmp_path_factory.mktemp("cronos-mempool")
+    cfg = Path(__file__).parent / "configs/long_timeout_commit.yaml"
+    yield from setup_custom_cronos(path, 26200, cfg)
+
+
+def test_mempool(cronos):
+    w3: Web3 = cronos.w3
+    filter = w3.eth.filter("pending")
+    assert filter.get_new_entries() == []
+
+    key_from = KEYS["validator"]
+    address_from = ADDRS["validator"]
+    address_to = ADDRS["community"]
+    gas_price = w3.eth.gas_price
+    cli = cronos.cosmos_cli(0)
+
+    # test contract
+    wait_for_new_blocks(cli, 1)
+    block_num_2 = w3.eth.get_block_number()
+    print(f"block number contract begin at height: {block_num_2}")
+    contract = deploy_contract(w3, CONTRACTS["Greeter"])
+    tx = contract.functions.setGreeting("world").buildTransaction()
+    signed = sign_transaction(w3, tx, key_from)
+    txhash = w3.eth.send_raw_transaction(signed.rawTransaction)
+    w3.eth.wait_for_transaction_receipt(txhash)
+    # check tx in mempool
+    new_txs = filter.get_new_entries()
+    assert txhash in new_txs
+
+    # wait block update
+    wait_for_new_blocks(cli, 1)
+    greeter_call_result = contract.caller.greet()
+    assert "world" == greeter_call_result
+
+    # check mempool
+    all_pending = w3.eth.get_filter_changes(filter.filter_id)
+    print(f"all pending tx hash after block: {all_pending}")
+    assert len(all_pending) == 0
+
+    # check transaction
+    block_num_0 = w3.eth.get_block_number()
+    print(f"block number start: {block_num_0}")
+    nonce_begin = w3.eth.get_transaction_count(address_from)
+
+    sended_hash_list = []
+    for i in range(5):
+        nonce = nonce_begin + i
```

Comment:
```suggestion
        nonce = w3.eth.get_transaction_count(address_from)
```

we need to do this to test the API behavior for the pending transactions.

Reply:
I did it before like this, I send tx in a loop, and it runs very quickly, if I get nonce from the API each time, the API will return the same nonce, so that send tx will fail.

Reply:
it might be a bug, or instability of the api.